ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalClient.java

```diff
@@ -37,14 +37,18 @@ public void onValidated(ValidateResponse validateResponse, String serverJWT) {
         });
     }
 
-    void continueLookup(FragmentActivity activity, ThreeDSecureResult threeDSecureResult, CardinalValidateReceiver validateReceiver) {
+    void continueLookup(FragmentActivity activity, ThreeDSecureResult threeDSecureResult, CardinalValidateReceiver validateReceiver) throws BraintreeException {
         ThreeDSecureLookup lookup = threeDSecureResult.getLookup();
         String transactionId = lookup.getTransactionId();
         String paReq = lookup.getPareq();
-        Cardinal.getInstance().cca_continue(transactionId, paReq, activity, validateReceiver);
+        try {
+            Cardinal.getInstance().cca_continue(transactionId, paReq, activity, validateReceiver);
+        } catch (NullPointerException e) {
+            throw new BraintreeException("Cardinal SDK Error.", e);
```

Comment:
Do we want to send an analytics event on this specific error? Or do we have observability into this already?

Reply:
We actually don't since the `CardinalClient` doesn't depend on `BraintreeClient`. We basically wrap Cardinal's API to encapsulate the dependency. We could potentially forward the client token to `ThreeDSecureActivity` to send a error events.